Failed integration `test_workspace_access_e2e`: AssertionError: Scope ACLs were not applied correctly

```
    def verify_applied_scope_acls(
        self, scope_name: str, migration_state: GroupMigrationState, target: Literal["backup", "account"]
    ):
        base_attr = "workspace" if target == "backup" else "backup"
        for mi in migration_state.groups:
            src_name = getattr(mi, base_attr).display_name
            dst_name = getattr(mi, target).display_name
            src_permission = self._secrets_support.secret_scope_permission(scope_name, src_name)
            dst_permission = self._secrets_support.secret_scope_permission(scope_name, dst_name)
>           assert src_permission == dst_permission, "Scope ACLs were not applied correctly"
E           AssertionError: Scope ACLs were not applied correctly
```

--- a/src/databricks/labs/ucx/workspace_access/scim.py
+++ b/src/databricks/labs/ucx/workspace_access/scim.py
@@ -32,13 +32,11 @@ class ScimSupport(AclSupport):
         return migration_state.is_id_in_scope(item.object_id)
 
     def get_crawler_tasks(self):
-        groups = self._get_groups()
-        with_roles = [g for g in groups if g.roles and len(g.roles) > 0]
-        with_entitlements = [g for g in groups if g.entitlements and len(g.entitlements) > 0]
-        for g in with_roles:
-            yield partial(self._crawler_task, g, "roles")
-        for g in with_entitlements:
-            yield partial(self._crawler_task, g, "entitlements")
+        for g in self._get_groups():
+            if g.roles and len(g.roles) > 0:
+                yield partial(self._crawler_task, g, "roles")
+            if g.entitlements and len(g.entitlements) > 0:
+                yield partial(self._crawler_task, g, "entitlements")
 
     # TODO remove after ES-892977 is fixed
     @retried(on=[DatabricksError])

